src/m365/spo/commands/listitem/listitem-batch-set.ts

```diff
@@ -0,0 +1,334 @@
+    const response: ClientSvcResponseContents = json[0];
+    if (response.ErrorInfo) {
+      throw response.ErrorInfo.ErrorMessage + ' - ' + response.ErrorInfo.ErrorValue;
+    }
+  }
+
+  private mapActions(index: number, row: any, fields: FieldDetails[], users: UserDetail[], systemUpdate?: boolean): [string, number] {
+    const objectPathId = index;
+    let actionString = '';
+
+    fields.forEach((field: FieldDetails) => {
+      switch (field.TypeAsString) {
+        case 'User':
+          const userDetail = users.find(us => us.email === row[field.InternalName])!;
+          actionString += `<Method Name="ParseAndSetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="String">${userDetail.id}</Parameter></Parameters></Method>`;
+        case 'UserMulti':
+          const userMultiString: string[] = row[field.InternalName].split(';').map((element: string) => {
+            const userDetail = users.find(us => us.email === element)!;
+            return `<Object TypeId="{c956ab54-16bd-4c18-89d2-996f57282a6f}"><Property Name="Email" Type="Null" /><Property Name="LookupId" Type="Int32">${userDetail.id}</Property><Property Name="LookupValue" Type="Null" /></Object>`;
+          });
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="Array">${userMultiString.join('')}</Parameter></Parameters></Method>`;
+          break;
+        case 'Lookup':
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter TypeId="{f1d34cc0-9b50-4a78-be78-d5facfcccfb7}"><Property Name="LookupId" Type="Int32">${row[field.InternalName]}</Property><Property Name="LookupValue" Type="Null"/></Parameter></Parameters></Method>`;
+          break;
+        case 'LookupMulti':
+          const lookupMultiString: string[] = row[field.InternalName].split(';').map((element: string) => {
+            return `<Object TypeId="{f1d34cc0-9b50-4a78-be78-d5facfcccfb7}"><Property Name="LookupId" Type="Int32">${element}</Property><Property Name="LookupValue" Type="Null" /></Object>`;
+          });
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="Array">${lookupMultiString.join('')}</Parameter></Parameters></Method>`;
+          break;
+        default:
+          actionString += `<Method Name="ParseAndSetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="String">${(<any>row)[field.InternalName].toString()}</Parameter></Parameters></Method>`;
+          break;
```

Comment:
I could tweak the `parseCsvToJson` method to make this work. What do you think?

Reply:
@martinlingstuyl  I've added this at the beginning of the `mapActions forEach`.
![image](https://github.com/pnp/cli-microsoft365/assets/9193870/77fbe629-6a6a-4ff8-ac70-fb649bc006a9)
This will empty the values for the specified columns. This will also stop the code from erroring on empty columns.

Also, I've noticed that the `parseCsvToJson` function will return the headers in the first value of the array, and we loop over the fields (retrieved from the first value of the array), thus including all the fields.

Reply:
Never mind... It seems that this would only work when using multiple rows. Not when updating only a single record.

Reply:
OK So I've rewritten the code of the `parseCsvToJson` method so that we, instead of looping over the matches in a line, will loop over **every** header value. This should theoretically resolve the issue.

Reply:
Very nice work @MathijsVerbeeck!